R/infillOptFocus.R

```diff
@@ -8,11 +8,35 @@
 # See infillOptCMAES.R for interface explanation.
 infillOptFocus = function(infill.crit, models, control, par.set, opt.path, design, iter, ...) {
   global.y = Inf
+  
+  allRequirements = extractSubList(par.set$pars, "requires", simplify = FALSE)
+  allUsedVars = unique(do.call(base::c, sapply(allRequirements, all.vars)))
```

Comment:
`base::all.vars` returning a vector of all variables used in the expression? Unfortunately it is not perfect (ideally I would want a list of all _free_ variables) but it is the best R is giving me as it seems.